src/CallbackList.php

```diff
@@ -27,22 +27,25 @@ public function add(callable $callback, ?string $name = null): void
         }
     }
 
+
     /**
      * Remove a named callback.
      *
-     * @param The name originally passed in add()
-     * @return True if successfully removed, false if not found
+     * @param string $name The name originally passed in add()
+     * @return bool True if successfully removed, false if not found
      */
     public function remove(string $name): bool
     {
-        if (!array_key_exists($name, $this->callbacks)) {
+        if (!\array_key_exists($name, $this->callbacks)) {
```

Comment:
This feels like overkill... do you want to refer to someone's advocacy of this style?

Reply:
Nope!  I think it's one of the Slevomat rules

Reply:
Let's pull that rule out for now.

Reply:
This appears to be the rationale... so maybe it's a good idea? https://veewee.github.io/blog/optimizing-php-performance-by-fq-function-calls/

I'll merge it...

Reply:
Part of my rationale for creating this pull request was to instigate a discussion within initially yourself and then the SilverStripe community about using existing tools to improve code quality.  I accept your criticism re previous comments but good to see you questioning them as above.  I'm happy to seed a discussion via the SilverStripe forums at some point in the near future if you see merit in this

Reply:
Yeah one thing to note is that I released this as a personally-maintained micropackage to avoid dumping it on the wider Silverstripe maintenance community.

I have some sway there, though ;-)

The bigger issue with coding standards on the Silverstripe project is introducing them without making breaking changes. In general, no coding standard is worth breaking an API for.

So, things like method name and type signature rules are really hard...

On this repo it's comparatively easier because it's recently made, very simple, and still at 0.1.

I'd welcome a discussion (probably as an issue on the framework repo) about what subset of slevomat *is* possible without breaking APIs though. In general "successfully get a few rules added to the travis build" is better than "start a big discussion about a massive set of rules that is too time-consuming to implement" ;-)

Phpstan is a current topic on a few open issues.